PXC-3026: Fix most of WITH_WSREP issues in the 8.0 branch

venkatesh-prasad-v opened the pull request:
https://jira.percona.com/browse/PXC-3026

* Introduces scripts under the check_src directory, which detect differences between the pxc WITH_WSREP=OFF source and PS. For the same upstream versions, there shouldn't be any changes except for the existing whitelists.

  The script also disables WITH_INNODB_DISALLOW_WRITES, as that's another 3rd party path not present in PS.

* Whitelisted product name changes, as PXC should be called PXC even without WITH_WSREP.

* Added WITH_WSREP if(n)defs everywhere where PXC added/deleted logic.

* Removed cosmetic differences compared to PS.

* Some of the clang-format changes were reverted. 

Developed-by: Zsolt Parragi <zsolt.parragi@percona.com>


Note:
The script currently produces the following output
```
============
Error: ./unittest/gunit/log_event_status_size-t.cc
============
116d115
<     qe.thd->variables.binlog_ddl_skip_rewrite = 1;
============
Error: ./sql/auth/sql_authentication.cc
============
4039,4045c4039
<       std::vector<std::string> external_roles;
<       if (strlen(mpvio.auth_info.external_roles) > 0) {
<         boost::algorithm::split(external_roles, mpvio.auth_info.external_roles,
<                                 boost::is_any_of(","));
<       }
< 
<       if (acl_user->user != nullptr && !external_roles.empty()) {
---
>       if (acl_user->user != nullptr) {
4054c4048,4051
<         for (const auto &role : external_roles) {
---
>         std::vector<std::string> roles;
>         boost::algorithm::split(roles, mpvio.auth_info.external_roles,
>                                 boost::is_any_of(","));
>         for (const auto &role : roles) {

```
1. ./unittest/gunit/log_event_status_size-t.cc - has been fixed in 8.0.32 by [commit](https://github.com/percona/percona-xtradb-cluster/commit/9fdcad17f4816c988a1a92a876974c56101f3af8)
2. ./sql/auth/sql_authentication.cc - has been fixed in 8.0.32 by [commit ](https://github.com/percona/percona-xtradb-cluster/commit/fcf47657ccf76516d5edf5f3bafd215e29fddb30)

venkatesh-prasad-v commented:
Jenkins: https://pxc.cd.percona.com/view/8.0%20parallel%20MTR/job/pxc-8.0-pipeline-parallel-mtr/1590/

Failing tests: [mysqld--help-notwin](https://pxc.cd.percona.com/view/8.0%20parallel%20MTR/job/pxc-8.0-pipeline-parallel-mtr/1590/testReport/ubuntu-jammy.Debug.main/main/mysqld__help_notwin/) Fixed later.

dutow commented:
The script and the whitelists are not part of this commit?

kamil-holubicki commented on sql/sql_insert.cc:
```diff
@@ -2685,7 +2685,7 @@ void Query_result_insert::abort_result_set(THD *thd) {
     transactional_table = table->file->has_transactions();
     if (thd->get_transaction()->cannot_safely_rollback(Transaction_ctx::STMT)) {
-#ifdef WIT_WSREP
+#ifdef WITH_WSREP
```
Ooops! Nice catch! I wonder what are side effects of fixing it ;)

kamil-holubicki commented on storage/innobase/srv/srv0start.cc:
```diff
@@ -2840,12 +2840,12 @@ void srv_start_threads(bool bootstrap) {
   /* Start the buffer pool dump/load thread, which will access spaces thus
         must wait for DDL recovery */
-#ifdef WTIH_WSREP
+#ifdef WITH_WSREP
```
Again, nice catch!

kamil-holubicki commented on storage/innobase/trx/trx0trx.cc:
```diff
@@ -1356,17 +1355,17 @@ static void trx_start_low(
 
-#if 0
 #ifdef WITH_WSREP
+#if 0
```
Maybe remove this dead code? Leave comment if still relevant.

venkatesh-prasad-v replied:
Done. Removed.

kamil-holubicki commented on storage/innobase/trx/trx0trx.cc:
```diff
@@ -2323,19 +2322,19 @@ void trx_commit(trx_t *trx) /*!< in/out: transaction */
 
   trx_commit_low(trx, mtr);
 
+#ifdef WITH_WSREP
```
Remove dead code?

venkatesh-prasad-v replied:
Done

Commit:
PXC-3026: Fix most of WITH_WSREP issues in the 8.0 branch

https://jira.percona.com/browse/PXC-3026

* Introduces scripts under the check_src directory, which detect
  differences between the pxc WITH_WSREP=OFF source and PS. For the same
  upstream versions, there shouldn't be any changes except for the
  existing whitelists.

  The script also disables WITH_INNODB_DISALLOW_WRITES, as that's
  another 3rd party path not present in PS.

* Whitelisted product name changes, as PXC should be called PXC even
  without WITH_WSREP.

* Added WITH_WSREP if(n)defs everywhere where PXC added/deleted logic.

* Removed cosmetic differences compared to PS.

Developed-by: Zsolt Parragi <zsolt.parragi@percona.com>
